docs(react-native): UI component - LobbyView (#811)
GetStream/stream-video-js: packages/react-native-sdk/docusaurus/docs/reactnative/04-ui-components/call/lobby-view.mdx

title: LobbyView
---

> LobbyView is a component that is designed to make your entrance and invite experience to the meeting user-friendly. This mostly contains all the necessary information about the meeting including:
> 

Reviewer: ```suggestion
LobbyView is a component that is designed to make your entrance and invite experience to a meeting call user-friendly. The purpose is to allow users to test their setup before entering the meeting call. This component mainly contains all the necessary information about the meeting, including:

```

> - The call details such as its id, etc.
> - Participants who are already in the meeting.
> - Control the pre-join audio/video mute status.
> - Handle the join button which is used to join the meeting.

Reviewer: ```suggestion
- The call details such as its id, etc.
- Participants who are already in the meeting.
- Controlling the audio/video mute status before joining the meeting.
- Exposes a handler for the join meeting button.
```

![Preview of the LobbyView component.](../../assets/04-ui-components/lobby-view.png)

## General usage

```tsx
import { useCallback } from 'react';
import {

>   Call,
>   LobbyView,
>   StreamCall,
>   useCall,

Reviewer: ```suggestion
  LobbyView,
  useCall,
```

} from '@stream-io/video-react-native-sdk';

const LobbyViewComponent = () => {
  let call = useCall();

  const onCallJoinHandler = useCallback(async () => {
    try {
      // Handle what should happen before joining the call. Eg: showing loader, etc.
      await call?.join({ create: true });
      // Handle what should happen after the call is joined. Eg: navigation, etc.
    } catch (error) {
      // Handle the case when there is an error while joining the call
    }

>   }, [call, appStoreSetState]);

Author: By mistake, let me remove it
